test/Microsoft.IdentityModel.Tokens.Tests/JsonWebKeySetTests.cs

```diff
@@ -121,6 +128,367 @@ public void GetSets()
+                    ExpectedSigningKeys = new List<SecurityKey>() { CreateRsaSecurityKey(jsonWebKeySet, 0) },
+                    ExpectedException = ExpectedException.NoExceptionExpected,
+                    TestId = "ValidRsaInvalidEcSkipUnresolved",
+                });
+
+                jsonWebKeySet = new JsonWebKeySet(DataSets.JsonWebKeySetOneValidRsaOneInvalidEcString);
+                theoryData.Add(new JsonWebKeySetTheoryData
+                {
+                    JsonWebKeySet = jsonWebKeySet,
+                    SkipUnresolvedJsonWebKeys = false,
+                    ExpectedSigningKeys = new List<SecurityKey>() { CreateRsaSecurityKey(jsonWebKeySet, 0), (jsonWebKeySet.Keys as List<JsonWebKey>)[1] },
+                    ExpectedException = ExpectedException.NoExceptionExpected,
+                    TestId = "ValidRsaInvalidEc",
+                });
+
+                jsonWebKeySet = DataSets.JsonWebKeySetX509Data;
+                theoryData.Add(new JsonWebKeySetTheoryData
+                {
+                    JsonWebKeySet = jsonWebKeySet,
+                    SkipUnresolvedJsonWebKeys = false,
+                    ExpectedSigningKeys = new List<SecurityKey>() { CreateX509SecurityKey(jsonWebKeySet, 0) },
+                    ExpectedException = ExpectedException.NoExceptionExpected,
+                    TestId = "ValidX5c",
+                });
+
+                jsonWebKeySet = new JsonWebKeySet(DataSets.JsonWebKeySetBadX509String);
+                theoryData.Add(new JsonWebKeySetTheoryData
+                {
+                    JsonWebKeySet = jsonWebKeySet,
+                    SkipUnresolvedJsonWebKeys = true,
+                    ExpectedSigningKeys = new List<SecurityKey>() { CreateRsaSecurityKey(jsonWebKeySet, 0) },
+                    ExpectedException = ExpectedException.NoExceptionExpected,
+                    TestId = "InvalidX5cSkipUnresolvedAddRsa",
+                });
+
+                jsonWebKeySet = new JsonWebKeySet(DataSets.JsonWebKeySetBadX509String);
+                theoryData.Add(new JsonWebKeySetTheoryData
+                {
+                    JsonWebKeySet = jsonWebKeySet,
+                    SkipUnresolvedJsonWebKeys = false,
+                    ExpectedSigningKeys = new List<SecurityKey>() { CreateRsaSecurityKey(jsonWebKeySet, 0), (jsonWebKeySet.Keys as List<JsonWebKey>)[0] },
+                    ExpectedException = ExpectedException.NoExceptionExpected,
+                    TestId = "InvalidX5c",
+                });
+
+                jsonWebKeySet = new JsonWebKeySet(DataSets.JsonWebKeySetOneInvalidEcOneValidEcString);
+                theoryData.Add(new JsonWebKeySetTheoryData
+                {
+                    JsonWebKeySet = jsonWebKeySet,
+                    SkipUnresolvedJsonWebKeys = true,
+                    ExpectedSigningKeys = new List<SecurityKey>(),
+                    ExpectedException = ExpectedException.NoExceptionExpected,
```

Comment:
nit: TheoryDataBase sets this to NoExceptionExpected by default.

Reply:
I was trying to be explicit in the previous iterations, but this is completely redundant as we are not throwing anymore. I'll remove it.

Reply:
addressed with d239fd43fc891e23202bd4532de76a4db05c9239